mvp-common/src/test/java/com/hannesdorfmann/mosby/mvp/MvpBasePresenterTest.java

```diff
@@ -35,4 +39,32 @@ public void testOnDestroy(){
   private void noViewAttached(){
     Assert.assertFalse(isViewAttached());
   }
+
+  @Test
+  public void testAttachView() throws Exception {
+    MvpBasePresenter<MvpView> presenter = new MvpBasePresenter<>();
+    assertFalse(presenter.isViewAttached());
+    presenter.attachView(new MvpView() {
+    });
+    assertTrue(presenter.isViewAttached());
+  }
+
+  @Test
+  public void testGetView() throws Exception {
+    MvpBasePresenter<MvpView> presenter = new MvpBasePresenter<>();
+    MvpView mvpView = new MvpView() {
+    };
+    presenter.attachView(mvpView);
+    assertEquals(mvpView, presenter.getView());
+  }
+
+  @Test
+  public void testDetachView() throws Exception {
+    MvpBasePresenter<MvpView> presenter = new MvpBasePresenter<>();
+    MvpView mvpView = new MvpView() {
+    };
+    presenter.attachView(mvpView);
+    presenter.detachView(false);
+    assertEquals(null, presenter.getView());
+  }
```

Comment:
Yes, I did these tests for an own patch to add annotations in my project. I minimally adapted them and contributed, but there could be a lot of overlap as I only later saw your tests for the same in master.